tests/centreon/plugins/statefile.t

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use Test::More;
+
+use FindBin;
+use lib "$FindBin::RealBin/../../../src";
+use centreon::plugins::statefile;
+
+# Mock options class
+{
+    package MockOptions;
+    sub new { bless {}, shift }
+    sub add_options { }
+    sub add_help { }
+}
+
+sub process_test {
+    my ($dir, $filename, $suffix, $format, $crypt_algo, $crypt_key) = @_;
+
+    # Create mock objects
+    my $options = MockOptions->new();
+    my $expected_statefile = $dir . '/' . $filename . $suffix;
+    unlink $expected_statefile if (-e $expected_statefile);
+
+    # Options to create a statefile object
+    $options->{statefile_dir}       = $dir;
+    $options->{statefile_suffix}    = $suffix;
+    $options->{statefile_format}    = $format;
+    $options->{statefile_cipher}    = $crypt_algo;
+    $options->{statefile_key}       = $crypt_key;
+
+    # Test object creation
+    my $statefile;
+    eval { $statefile = centreon::plugins::statefile->new(options => $options) };
+    ok(!$@, 'Object creation without errors');
+
+    $statefile->check_options(option_results => $options);
+
+    # Test if the object is blessed correctly
+    is(ref($statefile), 'centreon::plugins::statefile', 'Object is of correct class');
+
+    # Test if the object has the expected attributes
+    can_ok($statefile, qw(new check_options read write get get_string_content error));
+
+    # Try a first read to initiate a session
+    eval { $statefile->read(statefile => 'test_statefile') };
+    ok(!$@, 'Read method executed without errors');
+
+    # Test write method
+    eval { $statefile->write(data => { some_key => 'some_value' }) };
```

Comment:
Why don't you check the output of the method with is() instead of using eval ?
It look like the sub read() can only exit() if the option no_quit is set to 0, and write  never raise error ?